File: src/main/java/com/rubensminoru/partitioners/TimeBasedPartitioner.java
package com.rubensminoru.partitioners;

import com.rubensminoru.messages.KafkaMessage;
import com.rubensminoru.writers.Writer;

public class TimeBasedPartitioner implements Partitioner {
    public Long DURATION_MS = 3600000L;

    public boolean check(KafkaMessage message, Writer writer) {
        return ((int) (message.getTimestamp() / DURATION_MS)) == ((int) (writer.getTimestamp() / DURATION_MS));
    }
}

File: src/main/scala/com/rubensminoru/partitioners/TimeBasedPartitioner.scala
package com.rubensminoru.partitioners;

import com.rubensminoru.messages.KafkaAvroMessage;
import com.rubensminoru.writers.Writer;

class TimeBasedPartitioner extends Partitioner {
    def DURATION_MS = 3600000L

    def check(message:KafkaAvroMessage, writer:Writer): Boolean = {
        message.timestamp / DURATION_MS == (writer.getTimestamp / DURATION_MS) &&
          writer.isSchemaCompatible(message.schema)
    }
}
